manifests/agent/java.pp

```diff
@@ -0,0 +1,101 @@
+# == Class: newrelic::agent::java
+#
+# This class install the New Relic Java Agent
+#
+# Management of the JVM opts to make use of said agent are not supported
+# and must be managed by end users seperately
+#
+# Useful docs for reference
+# - https://docs.newrelic.com/docs/agents/java-agent/installation/install-java-agent
+# - https://docs.newrelic.com/docs/agents/java-agent/installation/include-java-agent-jvm-argument
+#
+# === Parameters:
+#
+# [*license_key*]
+#   NewRelic API license key (String)
+#
+# [*package_source_url*]
+#   URL for the newrelic zip. Used as the source for download of zip.
+#   Default: https://oss.sonatype.org/content/repositories/releases/com/newrelic/agent/java/newrelic-java/
+#
+# [*package_version*]
+#   Version of the newrelic java agent being used
+#   Default: 3.47.1
+#
+# [*install_dir*]
+#   Install dir for newrelic java agent.
+#   Default: /opt
+#
+# [*newrelic_application_name*]
+#   Name to give to your NewRelic application in the NewRelic GUI
+#   Default: webapp
+#
+# [*agent_user*]
+#   User for the newrelic agent
+#   Default: root
+#
+# [*agent_group*]
+#   Group for the newrelic agent
+#   Default: root
+#
+# [*manage_config*]
+#   Whether puppet should manage the newrelic.yml config file or not
+#   Default: true
+#
+class newrelic::agent::java (
+  String                                 $license_key,
+  String  $package_source_url            = 'https://oss.sonatype.org/content/repositories/releases/com/newrelic/agent/java/newrelic-java/',
+  String  $package_version               = '3.47.1',
+  String  $install_dir                   = '/opt',
+  String  $newrelic_application_name     = 'webapp',
+  String  $agent_user                    = 'root',
+  String  $agent_group                   = 'root',
+  Boolean $manage_config                 = true
+){
+
+  # == Package Installation
+  package{'unzip': ensure => installed}
```

Comment:
This should be split over multiple lines

Reply:
i think "should" is perhaps more of a "could" here. in any case it doesn't warrant a conversation and is done.